src/compiler/emitter.ts

```diff
@@ -570,9 +540,16 @@ var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, Promi
                 forEach(sourceFiles, emitSourceFile);
 
                 writeLine();
-                writeEmittedFiles(writer.getText(), jsFilePath, /*writeByteOrderMark*/ compilerOptions.emitBOM);
+
+                const sourceMappingURL = sourceMap.getSourceMappingURL();

```

Comment:
How about instead of this add a method on sourceMap with input TextWriter for JS file and that method does whatever is necessary?